vtm/src/org/oscim/utils/RTree.java

```diff
@@ -355,6 +369,70 @@ public boolean search(Box bbox, SearchCb<T> cb, Object context) {
         return true;
     }
 
+    class KnnItem implements Comparable<KnnItem> {
+        Branch<?> branch;
+        boolean isLeaf;
+        double squareDistance;
+
+        @Override
+        public int compareTo(KnnItem o) {
+            return Double.compare(squareDistance, o.squareDistance);
+        }
+    }
+
+    @Override
+    public List<T> searchKNearestNeighbors(Point center, int k, double maxDistance, List<T> results) {
+        // Refered https://github.com/mourner/rbush-knn/blob/master/index.js
+
+        if (results == null)
+            results = new ArrayList<T>(16);
+        results.clear();
+
+        PriorityQueue<KnnItem> queue = new PriorityQueue<>();
+        double maxSquareDistance = maxDistance * maxDistance;
```

Comment:
Cannot see in original code such action.
And what happens if max is infinity?

Reply:
The action is found in https://github.com/mourner/rbush-knn/pull/14 but seems to be not merged yet. In this current pull request `maxDistance * maxDistance` is done per children iteration, but it is clear that should be done only once before the iteration to improve performance.

`Double.POSITIVE_INFINITY * Double.POSITIVE_INFINITY` is `Double.POSITIVE_INFINITY`. Current unit tests are already using it.